### PR TITLE
Moved linear layer to constructor.

### PR DESCRIPTION
**Pull request by wiktorfilipiuk**

### CODE REVIEW

File: train.py [hunk 1]
```diff
@@ -137,6 +137,7 @@ def main(args):
         classification_mode=args.classification_mode,
         feature_extractor_name="cnn_extractor",
         transformer_name="hubert_transformer",
+        num_segments=8,
```
Comment (AgaMiko) [inline train.py:140]: I think it should be called num_frames and equal to `num_segments * frames_per_segment`. Also, please consider storing it as a variable as now it is used in two different places in code (model line 135 and dataset line 101)

File: models/feature_extractors/multi_frame_feature_extractor.py [hunk 2]
```diff
@@ -21,7 +23,7 @@ def forward(self, x):
 
         for frame in x:
             output = self.feature_extractor(frame).squeeze(0)
-            frame_features.append(output)
+            frame_features.append(Variable(output))
```
Comment (AgaMiko) [inline models/feature_extractors/multi_frame_feature_extractor.py:26]: label `output` as `Variable` so the optimizer knows it has to optimize it

File: models/model_loader.py [hunk 3]
```diff
@@ -14,7 +14,7 @@ class ModelLoader:
     }
 
     transformers_ = {
-        "vanilla_transformer": transformers.VanillaTransformer,
+        "fake_transformer": transformers.FakeTransformer,
```
Comment (AgaMiko) [inline models/model_loader.py:17]: rename `vanilla_transformer` to `fake_transformer` as it is describes the class better

File: models/transformers/fake_trasnformer.py [hunk 4]
```diff
@@ -3,7 +3,7 @@
 import torch.optim
 
 
-class VanillaTransformer(nn.Module):
+class FakeTransformer(nn.Module):
```
Comment (AgaMiko) [inline models/transformers/fake_trasnformer.py:6]: rename `vanilla_transformer` to `fake_transformer` as it is describes the class better

File: models/transformers/__init__.py [hunk 5]
```diff
@@ -1,2 +1,2 @@
-from .vanilla_trasnformer import VanillaTransformer
+from .fake_trasnformer import FakeTransformer
```
Comment (AgaMiko) [inline models/transformers/__init__.py:1]: rename `vanilla_transformer` to `fake_transformer` as it is describes the class better

File: models/transformers/hubert_transformer.py [hunk 6]
```diff
@@ -12,21 +13,35 @@ class HubertTransformer(nn.Module):
...
         Args:
             input_features (int): Argument to keep name convention consistency.
             output_features (int): Expected number of output features from the model.
+            num_segments (int): Number of frames associated with single sign.
         """
         super(HubertTransformer, self).__init__()
         configuration = HubertConfig()
         self.model = HubertModel(configuration)
-        self.model.feature_extractor._requires_grad = (
-            False  # fix require_grad error by blocking gradients
```
Comment (AgaMiko) [inline models/transformers/hubert_transformer.py:26]: remove temporary fix

File: datasets/dataset.py [hunk 7]
```diff
@@ -141,7 +141,12 @@ def _parse_annotationfile(self):
         ]
 
     def _sanity_check_samples(self):
-        for record in self.video_list:
+        for id, record in enumerate(self.video_list):
+            if any(isinstance(x, str) for x in record._data):
+                # Found datafile header. Removing it.
+                del self.video_list[id]
+                continue
```
Comment (AgaMiko) [inline datasets/dataset.py:148]: add check if text row from annotation file is a header

File: models/model.py [hunk 8]
```diff
@@ -86,7 +84,7 @@ def forward(self, input, **kwargs):
         x = self.multi_frame_feature_extractor(input)
         x = self.transformer(x)
         for head in self.cls_head:
-            predictions.append(head(x))
+            predictions.append(head(x.cpu()))
```
Comment (AgaMiko) [inline models/model.py:87]: move to cpu to fix training on GPU error

File: datasets/dataset.py [hunk 9]
```diff
@@ -141,7 +141,12 @@ def _parse_annotationfile(self):
         ]
 
     def _sanity_check_samples(self):
-        for record in self.video_list:
+        for id, record in enumerate(self.video_list):
+            if any(not x.isdigit() for x in record._data):
+                # Found datafile header. Removing it.
+                del self.video_list[id]
+                continue
```
Comment (AgaMiko) [inline datasets/dataset.py:148]: check if datafile row is a header (is not a digit)

File: train.py (hunk 1 above)
Comment (AgaMiko) [reply train.py]: fixed

**Review by AgaMiko — APPROVED** [review]

**Review by alicjak1519 — APPROVED** [review]
Nice work! ^^
